### DISCUSSION TITLE
Failure patterns across instances of servers.php

### DISCUSSION THREAD

**Discussion by mcfnord** [jamulussoftware/jamulus#3084] [Bug Investigation]
I've lost the discussion where @pljones asked about patterns in empty result sets from jamulus-php. I haven't comprehensively tested, but did test a failover technique that apparently reveals that [failures happen equally on geographically different instances](https://github.com/softins/jamulus-php/issues/6). I failover to one, and it failovers right back, until the interruption concludes. I still know very little. Here's a log that's kinda useless without timestamps (5 is SF, k is Tony's original UK instance):

5 choral
k choral
**5 classical
k classical
5 classical
k classical
5 classical
k classical
5 classical
k classical**
5 jazz
5 classical
k jazz
k classical
5 classical
k classical
5 jazz
k jazz
5 jazz
k jazz
5 jazz
5 classical
k jazz
k classical
5 choral
5 classical
k classical
5 classical
k classical
k choral
5 jazz
5 classical
k jazz
**k classical
5 classical
k classical
5 classical
k classical
5 classical
k classical
5 classical
k classical**
5 jazz
k jazz
5 jazz
k jazz
5 classical
k classical
5 classical
k classical
5 jazz
5 classical
k jazz
k classical
5 jazz
5 choral
k jazz
5 jazz
k jazz
5 classical
k classical
k choral
k anygenre1
k anygenre2
5 anygenre1
5 anygenre2
5 jazz
5 classical
k jazz
k classical
5 classical
k classical
5 classical
k classical
5 jazz
5 classical
k jazz
k classical
5 classical
k classical
5 jazz
5 classical
k jazz
k classical
5 classical
k classical
5 jazz
5 classical
k jazz
k classical

So it ping-pongs, and both instances seem to experience the same failure at the same time. I bet it's a timer in servers.php that times out, returning an empty set. But why? These failures certainly seemed to be instance-independent earlier, but this data strongly suggests consistency in the failures across instances. I suspect the one-offs are very brief failures that are resolved before the other instance is queried (about 5-10 seconds later). Obviously a much more comprehensive study of the question could be performed.

**Comment by ann0see** [comment]
I think Tony (softins) is the right person to judge that. Not sure who else has deep knowledge about Jamulus-php

**Reply by mcfnord** [reply]
The key observation here is that failure seems to happen across multiple instances at once. PLJones said that pattern would interest him. I plan to make a better test.
